torch/csrc/autograd/function.h

```diff
@@ -303,6 +316,7 @@ struct Function : std::enable_shared_from_this<Function> {
   // Since `Function`s are neither copyable nor moveable, we can have const
   // fields.
   const uint64_t sequence_nr_;
+  const int backwards_priority_;
```

Comment:
I can't see why do we need this. Since there's only one kind of special nodes, that should always have a priority greater than all other nodes, we can just assign `-1` to their `sequence_number`. This avoids both increasing the struct size, and complicating the scheduling logic.

Reply:
My original hack was actually to assign UINT_MAX to the sequence_nr to test the behavior, but I think this fix has comparable complexity and preserves the sequence_nr, and the sequence_nr is well-defined and may be independently useful later.

Even if we clobber the sequence_nr we still need a constructor to set it properly, so we only save the backwards_priority_ declaration, its getter, and the logic in the priority queue that checks it. This is only one check and one integer of savings at the cost of hacking up sequence_nr. 

Your call but I think the complexity is comparable and I like not hacking sequence_nr. Once you let me know which way you want to go I'll have the updated PR shortly.

Reply:
Yeah, I prefer @mruberry's approach of not clobbering `sequence_nr`.

Reply:
sequence_nr_ already has pretty weak consistency guarantees so I don’t think we should be using it for anything else than a hint for scheduling. Adding a single “infinite” point at the end of an order doesn’t seem too hacky for me, and it certainly makes the scheduling hint invariant easier

Reply:
Cool; if we're looking at sequence_nr_ only as a backwards hint that makes sense. I'll make the change and ensure the comments reflect this.